servers/lib/src/files/services/gitlab-files.service.ts

```diff
@@ -21,32 +21,35 @@ export default class GitlabFilesService implements IFilesService {
   }
 
   private async parseArguments(
-    path: string
+    path: string,
   ): Promise<{ domain: string; parsedPath: string }> {
     const gitlabGroup = this.configService.get("GITLAB_GROUP");
     const pathParts: string[] = path.split("/");
     const project: string = pathParts[0];
 
     // Only prepend the gitlabGroup if it's not already part of the path
     const domain: string =
-      project === gitlabGroup ? project : `${gitlabGroup  }/${  project}`;
+      project === gitlabGroup
+        ? `${project}/${pathParts[1]}`
+        : `${gitlabGroup}/${project}`;
 
-    const parsedPath = pathParts.slice(1).join("/");
+    const parsedPath =
+      project === gitlabGroup
+        ? pathParts.slice(2).join("/")
+        : pathParts.slice(1).join("/");
     return { domain, parsedPath };
   }
 
   private async sendRequest(query: string): Promise<Project> {
+    this.configService.get("GITLAB_TOKEN");
     try {
       const response = await axios({
         url: "https://gitlab.com/api/graphql",
         method: "post",
         headers: {
           "Content-Type": "application/json",
```

Comment:
Why does the gitlab accept the request without bearer token? Is it because the corresponding gitlab group `dtaas` public? The code should work even if the gitlab group is private. Without the gitlab token, fetching from private repository becomes difficult.

Reply:
Fixed